### DISCUSSION TITLE
How to set a different point color for each X Value on Chart?

### DISCUSSION THREAD

**Discussion by diegoup2** [patrykandpatrick/vico#799] [Questions]
### Question

Hello, thanks for the amazing library on Compose. Im currently working on a multi line graph with different color and shapes for some of the X values on the graph.

Something like this:

![Screenshot 2024-07-17 at 1 43 33 PM](https://github.com/user-attachments/assets/c2199346-5390-4391-ac7c-3539952c836c)

Im guessing by looking at the doc and code that I have to create my custom PointProvider?

Also, offtopic, I guess it will be the same for a VerticalBox decoation? Create the custom class and override the drawOverLayers? 

Thanks

### Vico version(s)

2.0.0-alpha.23

### UI framework(s)

Jetpack Compose

**Comment by patrickmichalik** [comment]
Hello! Thanks for the kind words. Yes to both questions. Regarding the second one, we plan on adding built-in `VerticalBox` and `VerticalLine` (#615) `Decoration`s in the future.
